django/db/backends/base/operations.py

```diff
@@ -792,3 +792,31 @@ def prepare_join_on_clause(self, lhs_table, lhs_field, rhs_table, rhs_field):
     def format_debug_sql(self, sql):
         # Hook for backends (e.g. NoSQL) to customize formatting.
         return sqlparse.format(sql, reindent=True, keyword_case="upper")
+
+    def format_json_path_numeric_index(self, num):
+        """
+        Hook for backends to customize indexing in JSON paths.
```

Comment:
Perhaps we could be a bit more explicit to avoid confusion with database indexes?

```suggestion
        Hook for backends to customize array indexing in JSON paths.
```